proto/redpanda/api/dataplane/v1alpha1/pipeline.proto

```diff
@@ -0,0 +1,212 @@
+    (buf.validate.field).required = true,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {description: "The minimum amount of resources needed for the pipeline to run."}
+  ];
+
+  // The maximum amount of resources that should be assigned to the pipeline
+  Resources limit = 6 [(grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {description: "The maximum amount of resources that should be assigned to the pipeline."}];
+
+  Status status = 7 [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  message Spec {}
+
+  message Resources {
+    string memory_shares = 1;
+    string cpu_shares = 2;
+  }
+
+  enum State {
+    STATE_UNSPECIFIED = 0;
+    STATE_STARTING = 1;
+    STATE_RUNNING = 2;
+    STATE_STOPPING = 3;
+    STATE_STOPPED = 4;
+    STATE_ERROR = 5;
+  }
+
+  message Status {
+    State state = 1;
+    string error = 2;
+  }
+}
+
+service PipelineService {
+  rpc CreatePipeline(CreatePipelineRequest) returns (CreatePipelineResponse);
+  rpc GetPipeline(GetPipelineRequest) returns (GetPipelineResponse);
+  rpc ListPipelines(ListPipelinesRequest) returns (ListPipelinesResponse);
+  rpc UpdatePipeline(UpdatePipelineRequest) returns (UpdatePipelineResponse);
+  rpc DeletePipeline(DeletePipelineRequest) returns (DeletePipelineResponse);
+  rpc StopPipeline(StopPipelineRequest) returns (StopPipelineResponse);
+  rpc StartPipeline(StartPipelineRequest) returns (StartPipelineResponse);
+}
+
+message PipelineCreate {
+  string display_name = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.pattern = "^[A-Za-z0-9-_ /]+$",
+    (buf.validate.field).string.min_len = 3,
+    (buf.validate.field).string.max_len = 128
+  ];
+  string description = 2 [(buf.validate.field).string.max_len = 1000];
+
+  string config_yaml = 3 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
+  ];
+  Pipeline.Resources request = 4 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
+  ];
+  Pipeline.Resources limit = 5;
+}
+
+message CreatePipelineRequest {
+  PipelineCreate pipeline = 1;
+}
+
+message CreatePipelineResponse {
+  Pipeline pipeline = 1;
+}
+
+message GetPipelineRequest {
+  string id = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.len = 20
+  ];
+}
+
+message GetPipelineResponse {
+  Pipeline pipeline = 1;
+}
+
+message ListPipelinesRequest {
+  Filter filter = 1;
+  int32 page_size = 2 [
+    (buf.validate.field).int32 = {
+      gte: -1,
+      lte: 1000
+    },
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "Limit the paginated response to a number of items. Defaults to 100. Use -1 to disable pagination.",
+      minimum: -1,
+      maximum: 1000
+    }
+  ];
+  string page_token = 3;
+
+  message Filter {
+    optional string display_name = 1;
```

Comment:
```suggestion
    string display_name = 1;
```

NIT: I think we don't need to differentiate between *string and string here no? `display_name` is a required field and searching for `display_name=""` doesn't make sense, so we can assume this as being unset. No strong opinion on this.

Reply:
This comes from:

See https://google.aip.dev/149

```
Important: Services should not need to distinguish between the default value and unset most of the time; if an alternative design does not require such a distinction, it is usually preferred. In practice, this means optional should only ever be used for integers and floats.
```

It's also done like this here: https://github.com/redpanda-data/cloudv2/blob/a00cbf359bff86374c23aa36ab6d90aa28fccd89/proto/public/cloud/redpanda/api/controlplane/v1beta2/cluster.proto#L1296

empty string is not a legit value, so it's preferable to use it for the field presence check.